### PR TITLE
Fix K8s/OpenShift deployment when using management interface+https

### LINKED ISSUE #32225: Quarkus-openshift extension fails to deploy application which use management interface+https
### Describe the bug

I have an application, which uses separate management interface[1] with HTTPS and deployed on Openshift via quarkus-openshift extension[2]. The application fails to start after the deployment.

[1] https://github.com/quarkusio/quarkus/pull/30506
[2] https://quarkus.io/guides/deploying-to-openshift

### Expected behavior

Quarkus openshift extension should deploy an application in a completely working state.

### Actual behavior

`error: update acceptor rejected openshift-quickstart-1: pods for rc 'fvd-management-https/openshift-quickstart-1' took longer than 600 seconds to become available` in the logs of deployment container and `  Warning  Unhealthy       3s (x5 over 43s)   kubelet            Startup probe failed: Get "http://10.128.2.196:9000/q/health/started": EOF` in the events (`oc describe pod openshift-quickstart-1-$whatever`) for the app itself

### How to Reproduce?

1. Clone: `git clone git@github.com:fedinskiy/reproducer.git -b openshift-extension-management-https`
2. Create new openshift project `oc new-project fvd-test-management-https`
3. Deploy the app according to manual: `mvn clean install -Dquarkus.kubernetes.deploy=true -Dquarkus.openshift.route.expose=true -Dquarkus.kubernetes-client.trust-certs=true -Dquarkus.management.ssl.certificate.key-store-file=META-INF/server.keystore
-Dquarkus.management.ssl.certificate.key-store-password=password
`
Log contains something like 
`[INFO] [io.quarkus.kubernetes.deployment.KubernetesDeployer] The deployed application can be accessed at: http://openshift-quickstart-fvd-management-https.apps.ocp4-12.rest.of.url`
4. Check the app
```
[fedinskiy@localhost reproducer]$ curl -v http://openshift-quickstart-fvd-management-https.apps.ocp4-12
<omitted for brevity>
> User-Agent: curl/7.85.0
< HTTP/1.0 503 Service Unavailable
```

For comparison: 
1. `oc new-project fvd-management`
2. `mvn clean install -Dquarkus.kubernetes.deploy=true -Dquarkus.openshift.route.expose=true -Dquarkus.kubernetes-client.trust-certs=true`
3. The application can be accessed: ` curl http://openshift-quickstart-fvd-management.apps.ocp4-12<omitted>` 

### Output of `uname -a` or `ver`

6.0.18-300.fc37.x86_64

### Output of `java -version`

17.0.5, vendor: GraalVM Community

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

34da5133fd7dd9997d1cc7d52f80bbf81cd27a17

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)

### Additional information

This is a follow-up to https://github.com/quarkusio/quarkus/issues/32135

### PR DESCRIPTION
**Pull request by Sgitario**
Note that these changes will move the management ssl configuration to built-time configuration. 

Moreover, the logic to select the HTTPS schema mimics to the logic in VertxHttpRecorder.initializeMainHttpServer method.

Fix https://github.com/quarkusio/quarkus/issues/32225

### CODE REVIEW

**Comment by gsmet** [comment]
Pinging @geoand too. Moving securty-related config to build time looks a bit impractical to me.

**Comment by geoand** [comment]
Unfortunately with all the stuff that is going on, I won't be able to look at this for at least 10 days

**Comment by cescoffier** [comment]
Let's discuss this after devoxx greece. While I agreed on reducing the flexibility, it can become problematic. Especially because I was also thinking of reducing the TLS configuration requirement (if you configure a cert, it's enabled). This approach may prevent these improvements.

**Comment by Sgitario** [comment]
> Let's discuss this after devoxx greece. While I agreed on reducing the flexibility, it can become problematic. Especially because I was also thinking of reducing the TLS configuration requirement (if you configure a cert, it's enabled). This approach may prevent these improvements.

Sure. All these arguments go into the direction of adding a new property `quarkus.management.ssl.certificate.enabled=true` to users enable it on. 
+1 Let's postpone it after the Devoxx greece.

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java [hunk 1]
```diff
@@ -8,12 +8,22 @@
...
+    public KubernetesProbePortNameBuildItem(String name, String schema) {
         this.name = name;
+        this.schema = schema;
     }
 
     public String getName() {
         return name;
     }
+
+    public String getSchema() {
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java]: ops! PR updated.

File: extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java [hunk 2]
```diff
@@ -466,4 +477,17 @@ public String updateApiUrl(String original, String healthPath) {
     private static boolean shouldInclude(LaunchModeBuildItem launchMode, SmallRyeHealthConfig healthConfig) {
         return launchMode.getLaunchMode().isDevOrTest() || healthConfig.ui.alwaysInclude;
     }
+
+    private static String selectSchemeForManagement() {
+        Config config = ConfigProvider.getConfig();
```
Comment (cescoffier) [inline extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java:482]: I would add a comment saying this is a best-effort approach. 
We also discussed a warning with @iocanel

**Review by cescoffier — APPROVED** [review]

File: extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java (hunk 2 above)
Comment (Sgitario) [reply extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java:482]: I understood that it should be INFO messages, not warnings. If so, these will be covered in a separate task (see the third point in epic https://github.com/quarkusio/quarkus/issues/33307).
I'll add a comment about the best-effort approach.

**Comment by Sgitario** [comment]
@iocanel can you have a look at this pull request please? Thanks!

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java [hunk 3]
```diff
@@ -36,6 +36,12 @@ public class ProbeConfig {
     @ConfigItem
     Optional<String> httpActionPath;
 
+    /**
+     * The schema of the {@literal HTTP get} action. Can be either "HTTP" or "HTTPS".
+     */
+    @ConfigItem
+    Optional<String> httpActionSchema;
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java:43]: Is this correct? Or it was meant to be `httpActionScheme`?

**Review by iocanel — APPROVED** [review]
lgtm

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java (hunk 3 above)
Comment (Sgitario) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java]: Ops, thanks for spotting! 
PR updated

**Comment by Sgitario** [comment]
I'm not sure if the failing tests are related or not. I will take a look asap.

**Comment by Sgitario** [comment]
I tried the test that failed locally, and it worked for me. 
@geoand @gsmet do we know if these tests are flaky?

**Comment by geoand** [comment]
It might be a case of us needing to increase the timeouts for the native tests

**Comment by Sgitario** [comment]
CI is green now. Merging!

**Pull request merged**